waiter/test/waiter/token_test.clj

```diff
@@ -2721,3 +2723,241 @@
+  "Assert that list of channels next message"
+  [chans msg]
+  `(let [chans# ~chans
+         msg# ~msg]
+     (assert-channels-next-message-with-fn chans# #(= % msg#))))
+
+(let [get-token-hash (fn [kv-store token] (sd/token-data->token-hash (kv/fetch kv-store token)))
+      get-latest-state (fn [query-chan]
+                         (let [temp-chan (async/chan)]
+                           (async/>!! query-chan temp-chan)
+                           (async/<!! temp-chan)))
+      create-watch-chans (fn []
+                           (for [_ (range 10)]
+                             (async/chan)))
+      add-watch-chans (fn [tokens-watch-channels-update-chan watch-chans]
+                        (doseq [chan watch-chans]
+                          (async/put! tokens-watch-channels-update-chan {:channel-event :add :watch-chan chan})))
+      remove-watch-chans (fn [tokens-watch-channels-update-chan watch-chans]
+                           (doseq [chan watch-chans]
+                             (async/put! tokens-watch-channels-update-chan {:channel-event :remove :watch-chan chan})))
+      stop-tokens-watch-maintainer (fn [go-chan exit-chan]
+                                     (async/>!! exit-chan :exit)
+                                     (async/<!! go-chan))
+      auth-user "auth-user"
+      token1-metadata {"cluster" "c1" "last-update-time" 1000 "owner" "owner1"}
+      token1-service-desc {"cpus" 1}
+      token1-index {:deleted false
+                    :last-update-time (get token1-metadata "last-update-time")
+                    :maintenance false
+                    :owner (get token1-metadata "owner")
+                    :token "token1"}]
+  (deftest test-start-tokens-watch-maintainer-empty-starting-state
+    (let [kv-store (kv/->LocalKeyValueStore (atom {}))
+          watch-chans (create-watch-chans)
+          exit-chan (async/chan 1)
+          tokens-update-chan (au/latest-chan)
+          tokens-watch-channels-update-chan (async/chan)
+          watch-refresh-timeout-ms 10000
+          {:keys [go-chan query-chan]}
+          (start-tokens-watch-maintainer kv-store tokens-update-chan tokens-watch-channels-update-chan watch-refresh-timeout-ms exit-chan)]
+      (is (= {:token->token-index {} :watches-count 0} (get-latest-state query-chan)))
+      (testing "watch-channels should receive empty list of tokens"
+        (add-watch-chans tokens-watch-channels-update-chan watch-chans)
+        (is (= {:token->token-index {} :watches-count 10} (get-latest-state query-chan)))
+        (assert-channels-next-message watch-chans '()))
+      (stop-tokens-watch-maintainer go-chan exit-chan)))
+  (deftest test-start-tokens-watch-maintainer-non-empty-starting-state
+    (let [kv-store (kv/->LocalKeyValueStore (atom {}))
+          watch-chans (create-watch-chans)
+          exit-chan (async/chan 1)
+          tokens-update-chan (au/latest-chan)
+          tokens-watch-channels-update-chan (async/chan)
```

Comment:
We can make the tests simpler, avoiding repeatedly initalizing the channels, if the channels are created inside the  `start-tokens-watch-maintainer`.